File: linked-list-find-loop/LinkedListFindLoop.js
class Node {
    constructor(value) {
        this.value = value;
        this.next = null;
        this.seen = false;
    }
}

exports.Node = Node;

var linkedListFindLoop = (head) => {
    var current = head;
    while(current != null && !current.seen) {
        current.seen = true;
        current = current.next;
    }

    return current;
}

exports.linkedListFindLoop = linkedListFindLoop;

File: linked-list-find-loop/LinkedListFindLoop.java
public class LinkedListFindLoop {
    static class Node {
        public Node next;
        public int value;
        public boolean seen = false;
        public Node(int value) {
            this.value = value;
        }
    }

    public Node findLoopLinkedList(Node head) {
        Node current = head;
        while (current != null && !current.seen ) {
            current.seen = true;
            current = current.next;
        }
        return current;
    }
}
